anndata/_core/anndata.py

```diff
@@ -832,34 +488,27 @@ def _init_as_actual(
         self._check_dimensions()
         self._check_uniqueness()
 
-        # raw
-        if raw is None:
-            self._raw = None
-        elif isinstance(raw, Raw):
-            self._raw = raw
-        else:
-            # is dictionary from reading the file, nothing that is meant for a user
-            if self.isbacked:
-                raw_key = "raw/X" if "raw/X" in self.file else "raw.X"
-                if raw_key not in self.file:
-                    raise KeyError(
-                        f"Tried to check shape of raw from {self.file.filename}, but there was no entry for 'raw/X'."
-                    )
-                if isinstance(self.file[raw_key], h5py.Group):
-                    shape = self.file[raw_key].attrs["shape"]
-                else:
-                    shape = self.file[raw_key].shape
-            else:
-                shape = raw["X"].shape
-
-            self._raw = Raw(
-                self,
-                X=raw.get("X", None),
-                var=_gen_dataframe(
-                    raw['var'], shape[1], ['var_names', 'col_names']
-                ),
-                varm=raw['varm'] if 'varm' in raw else None,
-            )
+        if self.filename:
+            assert not isinstance(
+                raw, Raw
+            ), "got raw from other adata but also filename?"
+            from ..readwrite.h5ad import _read_raw, _read_legacy_raw_h5ad
+
+            # Read raw dict from file
+            raw, raw_override = {}, raw
+            if "raw" in self.file:
+                raw = _read_raw(self.file)
+            _read_legacy_raw_h5ad(self.file, raw, self.filename)
+
```

Comment:
Doesn't this read `raw/X`? This is what we want to avoid in backed mode.

Reply:
Fixed.